clipper_admin/clipper_admin/docker/docker_container_manager.py

```diff
@@ -146,12 +181,19 @@ def start_clipper(self,
         query_cmd = ("--redis_ip={redis_ip} --redis_port={redis_port} "
                      "--prediction_cache_size={cache_size}").format(
                          redis_ip=self.redis_ip,
-                         redis_port=self.redis_port,
+                         redis_port=CLIPPER_INTERNAL_REDIS_PORT,
                          cache_size=cache_size)
-        query_labels = self.common_labels.copy()
-        query_labels[CLIPPER_QUERY_FRONTEND_CONTAINER_LABEL] = ""
+
         query_container_id = random.randint(0, 100000)
         query_name = "query_frontend-{}".format(query_container_id)
+        self.clipper_query_port = find_unbound_port(self.clipper_query_port)
+        self.clipper_rpc_port = find_unbound_port(self.clipper_rpc_port)
+        query_labels = self.common_labels.copy()
+        query_labels[CLIPPER_QUERY_FRONTEND_CONTAINER_LABEL] = ""
```

Comment:
Why is this label value empty?

Reply:
It's used to pin down specific container:

https://github.com/ucbrise/clipper/blob/1394de0867ac53efa5ed601bd91806d93d2f3ef4/clipper_admin/clipper_admin/docker/docker_container_manager.py#L211

Reply:
Hmm I'm still not sure I'm following. How exactly is this label used?

Reply:
It was used here: https://github.com/ucbrise/clipper/blob/062b968f96c3821a354374f0b90d9c3776618419/clipper_admin/clipper_admin/docker/docker_container_manager.py#L207-L218

to search for live query frontend containers. The value is empty simply because we don't need to add anything. "ai.clipper.query_frontend.label" is enough.

This was introduced from last summer's big refactoring: https://github.com/ucbrise/clipper/pull/251